components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -198,7 +205,19 @@ public Environment updateEnvironment(String tenantDomain, Environment environmen
 
         validateForUniqueVhostNames(environment);
         environment.setId(existingEnv.getId());
-        return apiMgtDAO.updateEnvironment(environment);
+        Environment updatedEnvironment = apiMgtDAO.updateEnvironment(environment);
+        // If the update is successful without throwing an exception
+        // Perform a separate task of updating gateway label names
+        updateGatewayLabelNameForGatewayPolicies(existingEnv.getDisplayName(), updatedEnvironment.getDisplayName(),
+                tenantDomain);
+        return updatedEnvironment;
+    }
+
+    private void updateGatewayLabelNameForGatewayPolicies(String oldLabel, String newLabel, String tenantDomain)
```

Comment:
method level description needed

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/1136a795360198388c6b4bd97029c20b0f898832

Thanks and Regards